### PR TITLE
Fix execution relay

### PR DESCRIPTION
**Pull request by yrong**
Fixes: SNO-577

### CODE REVIEW

File: relayer/relays/execution/main.go [hunk 1]
```diff
@@ -240,9 +272,10 @@ func (r *Relay) findEventsWithFilter(opts *bind.FilterOpts, paraID uint32, start
 			}
 			break
 		}
-
-		events = append(events, iter.Event)
-		if iter.Event.Nonce == start {
+		if iter.Event.Nonce >= start {
+			events = append(events, iter.Event)
+		}
+		if iter.Event.Nonce == start && opts.Start != 0 {
```
Comment (yrong) [inline relayer/relays/execution/main.go:278]: For `opts.Start == 0` is equivalent to the case `blockNumber < BlocksPerQuery` which is the first round of positive search, not stop at the first matching and just continue until all matching events found.

File: relayer/relays/execution/main.go [hunk 2]
```diff
@@ -97,6 +97,13 @@ func (r *Relay) Start(ctx context.Context, eg *errgroup.Group) error {
 				return err
 			}
 
+			if executionHeaderState.BlockNumber == 0 {
+				log.WithFields(log.Fields{
+					"channelId": r.config.Source.ChannelID,
+				}).Info("Beacon execution state syncing not started,wait...")
+				continue
+			}
+
```
Comment (claravanstaden) [inline relayer/relays/execution/main.go:106]: This is great, thanks for adding! On Goerli it takes quite a while and this message is much more clear than `no contract found at address`. 🙃

File: relayer/relays/execution/main.go (hunk 1 above)
Comment (claravanstaden) [reply relayer/relays/execution/main.go:278]: Thanks for clarifying! :)

File: relayer/relays/execution/main.go [hunk 3]
```diff
@@ -114,17 +121,42 @@ func (r *Relay) Start(ctx context.Context, eg *errgroup.Group) error {
...
+					"msgNonce":    ev.Nonce,
+					"address":     ev.Raw.Address.Hex(),
+					"blockHash":   ev.Raw.BlockHash.Hex(),
+					"blockNumber": ev.Raw.BlockNumber,
+					"txHash":      ev.Raw.TxHash.Hex(),
+					"dest":        ev.Dest,
+				})
+
+				if ev.Nonce <= paraNonce {
+					logger.Warn("inbound message outdated,just skipped")
```
Comment (claravanstaden) [inline relayer/relays/execution/main.go:145]:
```suggestion
					logger.Warn("inbound message outdated, just skipped")
```

File: relayer/chain/parachain/writer.go [hunk 4]
```diff
@@ -120,6 +121,8 @@ func (wr *ParachainWriter) WriteToParachainAndWatch(ctx context.Context, extrins
 				return fmt.Errorf("parachain write status was dropped, invalid, usurped or finality timed out")
 			}
 			if status.IsFinalized {
+				log.WithFields(log.Fields{
+					"extrinsic": extrinsicName, "block": status.AsFinalized}).Info("extrinsic finalized")
```
Comment (claravanstaden) [inline relayer/chain/parachain/writer.go:125]: Thinking this can be a debug log, but not major. 😄

**Review by claravanstaden — APPROVED** [review]

File: relayer/chain/parachain/writer.go (hunk 4 above)
Comment (yrong) [reply relayer/chain/parachain/writer.go:125]: Yes, just for more debug and the `blockHash` added here will be useful to search the block and extrinsic which error happens.

File: smoketest/tests/send_tokens.rs
Comment (vgeddes) [inline smoketest/tests/send_tokens.rs:1]: Alistair's got a working version of `send_tokens.rs` at https://github.com/Snowfork/snowbridge/pull/904/files#diff-11b438c85cff6243fa50f4d09bd3b95924e940c948e4be2b3effb85b43094617. So lets rather revert the changes to this file so there isn't a merge conflict

**Review by alistair-singh — APPROVED** [review]

File: core/packages/test/config/genesis.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
   "gasLimit": "80000000",
   "alloc": {
     "90A987B944Cb1dCcE5564e5FDeCD7a54D3de27Fe": {
-      "balance": "100000000000000000000"
+      "balance": "10000000000000000000000"
```
Comment (yrong) [inline core/packages/test/config/genesis.json:24]: Fund more here for smoke test.

**Review by vgeddes — APPROVED** [review]
+1
